pokemon_load_data.py

```diff
@@ -47,59 +46,51 @@ async def load_data():
 
     session = SessionLocal()
 
+    # Collect all existing Pokémon numbers from the database
+    existing_numbers = {pokemon.number for pokemon in session.query(Pokemon.number).all()}
+
+    # Collect Pokémon data for bulk insert
+    pokemon_data = []
+
     for entry in data:
         original_number = entry.get("#")
         number = original_number
 
         # Find the next available unique number
-        while session.query(Pokemon).filter_by(number=number).first() is not None:
+        while number in existing_numbers or any(p['number'] == number for p in pokemon_data):
             number += 1
```

Comment:
this method works, but I think this is not an efficient way.
Please refer to Namrata or Vishal code once, they have some efficient way for incrementing